src/overlays/actors/ovl_En_Floormas/z_en_floormas.c

```diff
@@ -91,144 +141,1104 @@ static DamageTable D_808D38D4 = {
+        !(this->actor.bgCheckFlags & 8)) {
+        func_808D1380(this, globalCtx);
+    } else if (this->actor.xzDistToPlayer > 400.0f) {
+        func_808D0CE4(this);
+    }
+}
+
+void func_808D108C(EnFloormas* this) {
+    s16 sp36 = this->unk_190 - this->actor.shape.rot.y;
+
+    this->actor.speedXZ = 0.0f;
+    if (sp36 > 0) {
+        SkelAnime_ChangeAnimTransitionStop(&this->skelAnime, &D_06002158, -3.0f);
+    } else {
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_06002158, -1.0f, SkelAnime_GetFrameCount(&D_06002158.common), 0.0f, 2,
+                             -3.0f);
+    }
+
+    if (this->actor.scale.x > 0.009f) {
+        this->unk_190 = sp36 * 0.06666667f;
+    } else {
+        this->skelAnime.animPlaybackSpeed *= 2.0f;
+        this->unk_190 = sp36 * 0.13333334f;
+    }
+
+    this->actionFunc = func_808D11BC;
+}
+
+void func_808D11BC(EnFloormas* this, GlobalContext* globalCtx) {
+    f32 frame;
+    f32 sp30;
+    f32 sp2C;
+
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime)) {
+        func_808D0C14(this);
+    }
+
+    if (((this->skelAnime.animPlaybackSpeed > 0.0f) && func_801378B8(&this->skelAnime, 21.0f)) ||
+        ((this->skelAnime.animPlaybackSpeed < 0.0f) && func_801378B8(&this->skelAnime, 6.0f))) {
+        if (this->actor.scale.x > 0.009f) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_FALL_WALK);
+        } else {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_FLOORMASTER_SM_WALK);
+        }
+    }
+
+    frame = this->skelAnime.animCurrentFrame;
+    if ((frame >= 7.0f) && (frame < 22.0f)) {
+        sp30 = Math_SinS(this->actor.shape.rot.y + 0x4268);
+        sp2C = Math_CosS(this->actor.shape.rot.y + 0x4268);
+        this->actor.shape.rot.y += this->unk_190;
+        this->actor.world.pos.x -= this->actor.scale.x * 2700.0f * (Math_SinS(this->actor.shape.rot.y + 0x4268) - sp30);
+        this->actor.world.pos.z -= this->actor.scale.x * 2700.0f * (Math_CosS(this->actor.shape.rot.y + 0x4268) - sp2C);
+    }
+}
+
+void func_808D1380(EnFloormas* this, GlobalContext* globalCtx) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06009520, 3.0f, 0.0f, SkelAnime_GetFrameCount(&D_06009520.common), 2,
+                         -3.0f);
+    this->actor.speedXZ = 0.0f;
+    this->actor.gravity = 0.0f;
+    func_808D08D0(this);
+    func_800BBDAC(globalCtx, &this->actor, &this->actor.world.pos, 15.0f, 6, 20.0f, 300, 100, 1);
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_FLOORMASTER_ATTACK);
+    this->actionFunc = func_808D1458;
+}
+
+void func_808D1458(EnFloormas* this, GlobalContext* globalCtx) {
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime)) {
+        func_808D161C(this);
+    }
+
+    this->actor.shape.rot.x += 0x140;
+    this->actor.world.pos.y += 10.0f;
+    Math_ApproachS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 3, 0xAAA);
+    Math_StepToS(&this->unk_192, 0x4B0, 0x64);
```

Comment:
```suggestion
    Math_StepToS(&this->unk_192, 1200, 100);
```
`this->unk_192` seems to be a pos